### DISCUSSION TITLE
Get HATCH to populate the entity boundary

### DISCUSSION THREAD

**Discussion by LongJun88** [mozman/ezdxf#671] [Q&A]
HI dear mozman：
How to read "HATCH" fill entity boundary, and obtain the corresponding boundary path coordinates?Thank you very much for giving me the test code.

**Comment by mozman** [comment]
The `HATCH` entity is very complex, here are the [docs](https://ezdxf.mozman.at/docs/dxfentities/hatch.html).

A good starting point to learn how to render the `HATCH` entity is the `Frontend.draw_hatch_entity()` method in the module [frontend.py](https://github.com/mozman/ezdxf/blob/master/src/ezdxf/addons/drawing/frontend.py) of the `drawing` add-on.

Some random topics you have to know:

- a `HATCH` can contain multiple separated hatched areas, where each area has an exterior path and holes and the holes can have holes and so on ...
- the boundary paths are in no particular order, not grouped by areas and are not tagged as exterior path or holes in any kind, the path flags `external`, `outermost` or `default` are not helpful to classifying the paths
- boundary paths can be degenerated, which means they do not represent a closed path
- the path orientation can be clockwise or counter-clockwise and has no special meaning

**Comment by LongJun88** [comment]
Mozman: Thank you very much. Let me take a look
